tutel/impls/communicate.py

```diff
@@ -75,7 +76,11 @@ def forward(ctx: Any, group: dist.ProcessGroup, input: Tensor):
             return input
         input = input.contiguous()
         output = torch.empty_like(input)
-        dist.all_to_all_single(output, input, group=group)
+        if AllToAllStatus.algo:
+            AllToAllStatus.init(group, 1, 0, input)
+            tutel_custom_kernel.all_to_all(output, input, AllToAllStatus.algo)
```

Comment:
No `group` specified?

Reply:
the first argument in `AllToAllStatus.init`, it will init using this group and store the communicator in `g_nccl_comm`, `tutel_custom_kernel.all_to_all` only uses the given `g_nccl_comm` and does not handle nccl init logic

in current design seems only one nccl comminutor is supported for alltoall, we can refine this in the future if needed